### ISSUE TITLE
Editor: Clicking Edit button on a non-site image errors & broken screen

### ISSUE THREAD

**Issue by jblz** [Automattic/wp-calypso#16259]
#### Steps to reproduce
1. Starting at URL: https://wordpress.com/post/:slug
2. Change to `HTML` mode
3. Embed an image that is not in the media library of the current site, e.g. something like:
```<img class="wp-image-7 size-full" src="https://twentysixteendemo.files.wordpress.com/2015/11/post.png" alt="post" width="1000" height="563" />```
4. Switch to `Visual` mode
5. Click the newly embedded image
6. Click the `Edit` button (pencil icon)

#### What I expected
To be able to edit the image using the Image Editor
...or, at least, the `Edit` button to not be shown if the image is "uneditable"

#### What happened instead
I was shown an "empty" media modal:
<img width="1085" alt="screen shot 2017-07-14 at 5 17 07 pm" src="https://user-images.githubusercontent.com/1587282/28231219-5001651c-68b8-11e7-8392-90d634ff4e94.png">

...& the console shows an error:
`Uncaught TypeError: Parameter 'url' must be a string, not boolean`

#### Browser / OS version
Chrome 59 / Mac Sierra

#### Screenshot / Video
Screencap: https://cloudup.com/cfgI2WniezG

**Comment by ramonjd** [comment]
I took a quick squiz at this and, while I can't reproduce the console.error, I'm getting some strange behaviour. 

Firstly, there's no check, when rendering the MediaModal, that any images exist at all. 

`this.props.view` === DETAIL, so it tries to open `<MediaModalDetail>` and load a non-existent image. 

```
	renderContent() {
		let content;

		switch ( this.props.view ) {
			case ModalViews.DETAIL:
				content = (
					<MediaModalDetail
						site={ this.props.site }
						items={ this.props.mediaLibrarySelectedItems }
						selectedIndex={ this.getDetailSelectedIndex() }
						onRestoreItem={ this.restoreOriginalMedia }
						onSelectedIndexChange={ this.setDetailSelectedIndex } />
				);
				break;
```
in `client/post-editor/media-modal/index.jsx`

`this.getDetailSelectedIndex()` goes away, tries to find the item in the `this.props.mediaLibrarySelectedItems` array, and returns 0, which is the default state. 

One quick fix would be to force the default case in the switch statement, which just shows the `<MediaLibrary />`, allowing the user to selected an already uploaded image. 

❓

**Comment by rachelmcr** [comment]
When I first tried to reproduce this issue, I tried pasting an image that I had copied from another site (rather than copy/pasting the HTML for an image) and ran into https://github.com/Automattic/wp-calypso/issues/21045. Since images pasted from other sites aren't given the `wp-image` class, that seems like a more likely bug to run into.

I also tried editing the default post "The Journey Begins" on a site (since that's where you'd naturally find this image), to try to reproduce the issue that way, but when I edited the image the editor loaded (with an upload date that matched my site's creation date):

![screenshot 2018-08-28 16 51 41](https://user-images.githubusercontent.com/8658164/44734805-ac9b7480-aae2-11e8-9672-3525edf9cd97.png)

I did reproduce the issue by copying the image html into an entirely new post, and then trying to edit the image, but that seems like a much less likely scenario given that the HTML has to include the `wp-image` class for this bug to occur.
